src/algebra/algebra/linear_algebra.lean

```diff
@@ -0,0 +1,157 @@
+/-
+Copyright (c) 2018 Kenny Lau. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kenny Lau, Yury Kudryashov
+-/
+import algebra.algebra.basic
+import linear_algebra.tensor_product
+
+/-!
+# Facts about algebras involving linear maps and tensor products
+
+We move a few basic statements about algebras out of `algebra.algebra.basic`,
+in order to avoid importing `linear_algebra.tensor_product` unnecessarily.
+-/
+
+universes u v w
+
+namespace algebra
+
+open_locale tensor_product
+open module
+
+section
+
+variables (R A : Type*) [comm_semiring R] [semiring A] [algebra R A]
+
+/-- The multiplication in an algebra is a bilinear map.
+
+A weaker version of this for semirings exists as `add_monoid_hom.mul`. -/
+def lmul : A →ₐ[R] (End R A) :=
+{ map_one' := by { ext a, exact one_mul a },
+  map_mul' := by { intros a b, ext c, exact mul_assoc a b c },
+  map_zero' := by { ext a, exact zero_mul a },
+  commutes' := by { intro r, ext a, dsimp, rw [smul_def] },
+  .. (show A →ₗ[R] A →ₗ[R] A, from linear_map.mk₂ R (*)
+  (λ x y z, add_mul x y z)
+  (λ c x y, by rw [smul_def, smul_def, mul_assoc _ x y])
+  (λ x y z, mul_add x y z)
+  (λ c x y, by rw [smul_def, smul_def, left_comm])) }
+
+variables {R A}
+
+@[simp] lemma lmul_apply (p q : A) : lmul R A p q = p * q := rfl
+
+instance linear_map.module' (R : Type u) [comm_semiring R]
+  (M : Type v) [add_comm_monoid M] [module R M]
+  (S : Type w) [comm_semiring S] [algebra R S] : module S (M →ₗ[R] S) :=
+{ smul := λ s f, linear_map.llcomp _ _ _ _ (algebra.lmul R S s) f,
+  one_smul := λ f, linear_map.ext $ λ x, one_mul _,
+  mul_smul := λ s₁ s₂ f, linear_map.ext $ λ x, mul_assoc _ _ _,
+  smul_add := λ s f g, linear_map.map_add _ _ _,
+  smul_zero := λ s, linear_map.map_zero _,
+  add_smul := λ s₁ s₂ f, linear_map.ext $ λ x, add_mul _ _ _,
+  zero_smul := λ f, linear_map.ext $ λ x, zero_mul _ }
+
+variables (R)
+
+/-- The multiplication on the left in an algebra is a linear map. -/
+def lmul_left (r : A) : A →ₗ[R] A :=
+lmul R A r
+
+/-- The multiplication on the right in an algebra is a linear map. -/
+def lmul_right (r : A) : A →ₗ[R] A :=
+(lmul R A).to_linear_map.flip r
+
+/-- Simultaneous multiplication on the left and right is a linear map. -/
+def lmul_left_right (vw: A × A) : A →ₗ[R] A :=
+(lmul_right R vw.2).comp (lmul_left R vw.1)
+
+lemma commute_lmul_left_right (a b : A) :
+  commute (lmul_left R a) (lmul_right R b) :=
+by { ext c, exact (mul_assoc a c b).symm, }
+
+/-- The multiplication map on an algebra, as an `R`-linear map from `A ⊗[R] A` to `A`. -/
+def lmul' : A ⊗[R] A →ₗ[R] A :=
+tensor_product.lift (lmul R A).to_linear_map
+
+variables {R A}
+
+@[simp] lemma lmul'_apply {x y : A} : lmul' R (x ⊗ₜ y) = x * y :=
+by simp only [algebra.lmul', tensor_product.lift.tmul, alg_hom.to_linear_map_apply, lmul_apply]
```

Comment:
Why not just move these two statements to `tensor_product` or perhaps `ring_theory.tensor_product` instead? Does any of the rest need tensor products?

Reply:
Ah, I was actually hoping to remove the import of `linear_algebra.bilinear_map` from `algebra.algebra.basic`. I've renamed this file, and updated the module-doc, to reflect this intent.